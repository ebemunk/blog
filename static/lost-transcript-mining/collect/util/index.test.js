import fs from 'fs'

import Promise from 'bluebird'
import sinon from 'sinon'
import axios from 'axios'

import {
	readFile,
	writeFile,
	download,
	insertObj
} from './'

const sandbox = sinon.sandbox.create()

describe('util', () => {
	afterEach(() => sandbox.restore())

	describe('readFile', () => {
		beforeEach(() => {
			sandbox.stub(fs, 'readFile').yields()
		})

		it('returns Promise', () => {
			const p = readFile('lol.bmp')
			expect(p).toBeInstanceOf(Promise)
		})

		it('calls fs.readFile with utf-8 encoding', () => {
			readFile('hehe.jpg')
			sinon.assert.calledWith(fs.readFile, 'hehe.jpg', 'utf-8')
		})
	})

	describe('writeFile', () => {
		beforeEach(() => {
			sandbox.stub(fs, 'writeFile').yields()
		})

		it('returns Promise', () => {
			const p = writeFile('derp.txt', 'souper sekrit')
			expect(p).toBeInstanceOf(Promise)
		})

		it('calls fs.writeFile', () => {
			writeFile('meh/heh', 'okay')
			sinon.assert.calledWith(fs.writeFile, 'meh/heh', 'okay')
		})
	})

	describe('download', () => {
		let sandbox

		beforeEach(() => {
			sandbox = sinon.sandbox.create()
		})

		afterEach(() => sandbox.restore())

		it('should write html to path', async () => {
			sandbox.stub(axios, 'get')
			.returns(Promise.resolve({
				data: '<html>yezs</html>'
			}))
			const writeMock = sandbox.stub(fs, 'writeFile')
			.callsArgWith(2)

			await download('url', 'path/')

			expect(writeMock.calledOnce).toBe(true)
			expect(writeMock.firstCall.args).toMatchSnapshot()
		})
	})

	describe('insertObj', () => {
		it('provide correct sql', () => {
			expect(insertObj({
				faf: 'etet',
				lolol: 'yaya',
				directions: [
					'some', 'things'
				]
			})).toMatchSnapshot()
		})
	})
})
