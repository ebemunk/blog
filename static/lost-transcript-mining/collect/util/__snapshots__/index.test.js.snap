// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util download should write html to path 1`] = `
Array [
  "path/",
  "<html>yezs</html>",
  [Function],
]
`;

exports[`util insertObj provide correct sql 1`] = `
Array [
  "
		INSERT INTO dialog
		(faf,lolol,directions)
		VALUES
		($1,$2,$3)
	",
  Array [
    "etet",
    "yaya",
    Array [
      "some",
      "things",
    ],
  ],
]
`;
